### PR TITLE
240719 Implemented move_one_date

### PR DESCRIPTION
**Pull request by rlagkwns0803**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
 
-def move_one_date():
-    # 1. 뒤로 가기
-    # 2. 다음 날짜 찾기
-    # 2-1. 다음이 10 이하면 article 누러주기
-    # 2-2. 10초과면: 다음페이지 넘어가기 버튼 누르고, 혀냊 page는 1번으로 초기화
-    return
+def move_one_date(driver):
+    next_button = driver.find_element(By.XPATH, '/html/body/div[1]/div/div/div/div/main/nav/div/div/a')
+    next_button.click()
```
Comment (rlagkwns0803) [inline main.py:48]: open the site by using XPATH and .click()

File: main.py [hunk 2]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
+
+    try:
+        start_date = datetime.strptime(start_date_str, date_format)
+        end_date = datetime.strptime(end_date_str, date_format)
+
+        # 두 날짜 간의 차이 계산
+        delta = end_date - start_date
+
+        # delta.days를 통해 일 수를 구할 수 있음
+        return delta.days + 1
```
Comment (rlagkwns0803) [inline main.py:64]: calculate volume of dates to repeat the 'move_one_date' through subtract two dates and plus one to include the end date

File: main.py [hunk 3]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
 
+    except ValueError:
+        return "올바른 날짜 형식이 아닙니다."
```
Comment (rlagkwns0803) [inline main.py:67]: comment error in wrong situation

File: main.py [hunk 4]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
-def parseData(startDate, endDate, unit):
+
+def parse_data(starting_date, ending_date, unit):
     # startDate, endDate
     # unit: monthly, weekly, yearly
-    date = startDate
 
-    # init driver and open rorkr
+    # open Page
+    print('[DBG] Opening rorkr')
```
Comment (rlagkwns0803) [inline main.py:75]: print certain sentence to find out where the problem occurs

File: main.py [hunk 5]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
     driver = initialize_driver()
     url = 'http://rorkr.com/'  # 데이터를 추출할 웹사이트의 URL
     driver.get(url)
 
+    FIRST_PAGE_TAG = '/html/body/div[1]/div/div/div/div/main/article[1]/header/h3/a'
```
Comment (rlagkwns0803) [inline main.py:80]: make a function for certain page in the site above ( rorkr.com )

File: main.py [hunk 6]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
+
+    # Wait for page to open
+    WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.XPATH, FIRST_PAGE_TAG))
```
Comment (rlagkwns0803) [inline main.py:84]: make driver to wait until find the location of FIRST_PAGE_TAG

File: main.py [hunk 7]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
+    )
+
+    # open first page
+    first_page_btn = driver.find_element(By.XPATH, FIRST_PAGE_TAG)
+    first_page_btn.click()
```
Comment (rlagkwns0803) [inline main.py:89]: open the page in FIRST_PAGE_TAG

File: main.py [hunk 8]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
+
+    date_cnt = days_between_dates(starting_date, ending_date)
+    print('[DBG] date_cnt is ' + str(date_cnt))
```
Comment (rlagkwns0803) [inline main.py:92]: identify the number of dates that has to repeat through 'days_between_dates'

File: main.py [hunk 9]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
+    
+
     # repeat for several pages
-    while date < endDate:  # 1. 하준
-        info = parse_one_page(driver)  # 2. 한휘
-        save_to_excel(info, unit)  # 3. 동혁. unit은 기본 1달치, 첫 도전은 1달치
+    while date_cnt > 0:
+        time.sleep(3)  # [TODO] Delete after debugging is done
```
Comment (rlagkwns0803) [inline main.py:97]: repeat code below according to date_cnt and waiting for second before next process

File: main.py [hunk 10]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
 
-        move_one_date()
-        date += 1
-    return
+        # info = parse_one_page()  # 2. 한휘
+        # save_to_excel(info, unit)  # 3. 동혁. unit은 기본 1달치, 첫 도전은 1달치
 
+        move_one_date(driver)
+        date_cnt -= 1
```
Comment (rlagkwns0803) [inline main.py:103]: move to next date and subtract one from entire dates to prevent the code works infinitely

File: main.py [hunk 11]
```diff
@@ -38,36 +43,69 @@ def save_to_excel(info):  # 동혁
...
+        print('[DBG] date_cnt is ' + str(date_cnt))
 
-# parseData("", "", "")
 
+parse_data("2024 07 19", "2024 07 20", "")
```
Comment (rlagkwns0803) [inline main.py:107]: test

File: main.py [hunk 12]
```diff
@@ -3,12 +3,17 @@
 from selenium.webdriver.chrome.service import Service
 from selenium.webdriver.chrome.options import Options
 from webdriver_manager.chrome import ChromeDriverManager
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+from datetime import datetime
+import time
```
[the remark this answers is not in the document]
Comment (DHCHO-pt) [reply main.py:10]: review.

File: main.py [hunk 13]
```diff
@@ -3,12 +3,17 @@
...
 
 
 def initialize_driver():
     # 크롬 웹드라이버 설정
     chrome_options = Options()
-    chrome_options.add_argument("--headless")  # 브라우저 창을 열지 않고 실행하기 위한 옵션
+    # chrome_options.add_argument("--headless")  # 브라우저 창을 열지 않고 실행하기 위한 옵션
```
[the remark this answers is not in the document]
Comment (DHCHO-pt) [reply main.py:16]: reviewed

File: main.py (hunk 1 above)
Comment (DHCHO-pt) [reply main.py:48]: reviewed

File: main.py (hunk 2 above)
Comment (DHCHO-pt) [reply main.py:64]: reviewed

File: main.py (hunk 3 above)
Comment (DHCHO-pt) [reply main.py:67]: reviewed

File: main.py (hunk 7 above)
Comment (DHCHO-pt) [reply main.py:89]: reviewed

File: main.py (hunk 8 above)
Comment (DHCHO-pt) [reply main.py:92]: reviewed

File: main.py (hunk 9 above)
Comment (DHCHO-pt) [reply main.py:97]: reviewed

File: main.py (hunk 10 above)
Comment (DHCHO-pt) [reply main.py:103]: reviewed

File: main.py (hunk 11 above)
Comment (DHCHO-pt) [reply main.py:107]: reviewed

**Pull request merged**